### ISSUE TITLE
ClientCredentialsTokenHandler doesn't work with all IDPs due to use of token_type for auth scheme

### ISSUE THREAD

**Issue by fultke01** [DuendeSoftware/Support#777]
**Which version of Duende.AccessTokenManagement are you using?**
Duende.AccessTokenManagement 2.0.2

**Which version of .NET are you using?**
net6.0

**Describe the bug**
The ClientCredentialsTokenHandler from Duende.AccessTokenManagement doesn't work with some IDPs' oauth 2.0 bearer token implementations due to behavior changes supporting the new draft DPoP spec.

The issue is that the handler maps the token_type to the auth header scheme, and a small number of IDPs return "bearer" instead of "Bearer" (e.g. [Facebook](https://developers.facebook.com/docs/pages/access-tokens/),  [MuleSoft ](https://docs.mulesoft.com/api-manager/2.x/oauth-grant-types-about#http-request-example-against-the-provider-to-get-a-token)). 

In my reading of RFC6750, it's very clear that the auth header scheme be only "Bearer" ([ref](https://datatracker.ietf.org/doc/html/rfc6750#section-3)), but I don't see any such statement about token_type or that it should be mapped to the scheme. Their implementation choices are inconvenient, I don't think they're out of compliance. After reading the draft for DPoP ([ref](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop#figure-7)) where it talks about possibly returning a different token_type than asked for, I can see why the new handler is implemented this way.

The older IdentityModel.AspNetCore implementation doesn't have this problem because it uses the hard coded string "Bearer" for the scheme.

I'm not sure if it makes more sense to simply validate that the token_type is a known scheme before using it, or do a more involved split of DPoP vs Bearer token handling.
https://github.com/DuendeSoftware/Duende.AccessTokenManagement/blob/951f552f4f1ed219c94b536af59c4fd74b2d9bea/src/Duende.AccessTokenManagement/AccessTokenHandler.cs#L93

**Comment by josephdecock** [comment]
@brockallen, I think you might want to take a look at this.

**Comment by brockallen** [comment]
Yes, rfc6749 says the "token_type" is case insensitive, but rfc6759 says "Bearer" (case sensitive). Fun.

We'll do something to parse the "token_type" response value to map it to "Bearer" (or "DPoP"). Thanks.
